numpy/core/tests/test_simd.py

```diff
@@ -501,6 +501,34 @@ def test_special_cases(self):
         nnan = self.notnan(self.setall(self._nan()))
         assert nnan == [0]*self.nlanes
 
+    import operator
+
+    @pytest.mark.parametrize('py_comp,np_comp', [
+        (operator.lt, "cmplt"),
+        (operator.le, "cmple"),
+        (operator.gt, "cmpgt"),
+        (operator.ge, "cmpge"),
+        (operator.eq, "cmpeq"),
+        (operator.ne, "cmpneq")
+    ])
+    def test_comparison_with_nan(self, py_comp, np_comp):
+        pinf, ninf, nan = self._pinfinity(), self._ninfinity(), self._nan()
```

Comment:
Hmmm, I have a question after all.  This seems like it tests float64 only?  Whether or not we add more tests, I think this should cover all tests.

Or... add generic float tests to `test_umath.py` if it doesn't fit here?

Reply:
It also covers float32

```numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[lt-cmplt] PASSED                                                                                                                                [ 83%] 
numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[le-cmple] PASSED                                                                                                                                [ 83%] 
numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[gt-cmpgt] PASSED                                                                                                                                [ 83%] 
numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[ge-cmpge] PASSED                                                                                                                                [ 83%] 
numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[eq-cmpeq] PASSED                                                                                                                                [ 83%] 
numpy/core/tests/test_simd.py::Test_SIMD_FP_128_baseline_f32::test_comparison_with_nan[ne-cmpneq] PASSED    ```

Reply:
Ah, thanks... inheritance magic I guess, sorry for the noise.